burn-tensor/src/tests/module/maxpool1d.rs

```diff
@@ -0,0 +1,98 @@
+#[burn_tensor_testgen::testgen(module_max_pool1d)]
+mod tests {
+    use super::*;
+    use burn_tensor::module::{max_pool1d, max_pool1d_with_indices};
+    use burn_tensor::{backend::Backend, Data, Tensor};
+
+    type IntElem = <TestBackend as Backend>::IntElem;
+
+    #[test]
+    fn test_max_pool1d_simple() {
+        let batch_size = 2;
+        let channels_in = 2;
+        let kernel_size = 3;
+        let padding = 1;
+        let stride = 1;
+
+        let x = TestTensor::from_floats([[
+            [0.9861, 0.5474, 0.4477, 0.0732, 0.3548, 0.8221],
+            [0.8148, 0.5474, 0.9490, 0.7890, 0.5537, 0.5689],
+        ]]);
+        let y = TestTensor::from_floats([[
+            [0.9861, 0.9861, 0.5474, 0.4477, 0.8221, 0.8221],
+            [0.8148, 0.9490, 0.9490, 0.9490, 0.7890, 0.5689],
+        ]]);
+
+        let output = max_pool1d(x, kernel_size, stride, padding);
+
+        y.to_data().assert_approx_eq(&output.into_data(), 3);
+    }
+
+    #[test]
+    fn test_max_pool1d_different_padding_stride_kernel() {
+        let batch_size = 1;
+        let channels_in = 1;
+        let kernel_size = 3;
+        let padding = 1;
+        let stride = 2;
+
+        let x = TestTensor::from_floats([[[0.6309, 0.6112, 0.6998, 0.4708]]]);
+        let y = TestTensor::from_floats([[[0.6309, 0.6998]]]);
+
+        let output = max_pool1d(x, kernel_size, stride, padding);
+
+        y.to_data().assert_approx_eq(&output.into_data(), 3);
+    }
+
+    #[test]
+    fn test_max_pool1d_with_neg() {
+        let batch_size = 1;
+        let channels_in = 1;
+        let kernel_size = 3;
+        let padding = 1;
+        let stride = 1;
+
+        let x = TestTensor::from_floats([[[-0.6309, -0.6112, -0.6998, -0.4708]]]);
+        let y = TestTensor::from_floats([[[-0.6112, -0.6112, -0.4708, -0.4708]]]);
+
+        let output = max_pool1d(x, kernel_size, stride, padding);
+
+        y.to_data().assert_approx_eq(&output.into_data(), 3);
+    }
+
+    #[test]
+    fn test_max_pool1d_with_indices() {
+        let batch_size = 1;
+        let channels_in = 1;
+        let kernel_size = 2;
+        let padding = 1;
+        let stride = 1;
+
+        let x = TestTensor::from_floats([[[0.2479, 0.6386, 0.3166, 0.5742]]]);
+        let indices = Data::<IntElem, 3>::from([[[0, 4, 4, 12, 12]]]);
```

Comment:
I also found this indices weird, probably caused by reshaping...

Reply:
It doesn't seem right, can you validate the test case with PyTorch?